### ISSUE TITLE
Bind cuDNN dropout initialization from ATen

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#5138]
In `torch/backends/cudnn/rnn.py` we have the last vestiges of the Python-side cuDNN bindings:

```
class Unserializable(object):

    def __init__(self, inner):
        self.inner = inner

    def get(self):
        return self.inner

    def __getstate__(self):
        # Note: can't return {}, because python2 won't call __setstate__
        # if the value evaluates to False
        return "<unserializable>"

    def __setstate__(self, state):
        self.inner = None


def init_dropout_descriptor(handle, dropout, train, dropout_seed, dropout_state):
    dropout_desc_name = 'desc_' + str(torch.cuda.current_device())
    dropout_p = dropout if train else 0
    if (dropout_desc_name not in dropout_state) or (dropout_state[dropout_desc_name].get() is None):
        dropout_state[dropout_desc_name] = Unserializable(
            cudnn.DropoutDescriptor(handle, dropout_p, dropout_seed)
        )
    dropout_desc = dropout_state[dropout_desc_name].get()
    dropout_desc.set_dropout(dropout_p, dropout_seed)
    return dropout_desc


def get_dropout_state(fn, handle):
    dropout_desc_name = 'desc_' + str(torch.cuda.current_device())
    dropout_p = fn.dropout if fn.train else 0
    dropout_desc = fn.dropout_state[dropout_desc_name].get()
    return dropout_desc.state
```

We'd like to get rid of this code entirely, and that means we need to allocate the tensor state in `dropout_desc` from ATen.

The code to do this already exists in `aten/src/ATen/cudnn/Descriptors.h`; it's just a matter of taking dropout parameters and returning an initialized dropout state buffer, which can be passed to subsequent ATen calls.

**Comment by apaszke** [comment]
We can't remove the `Unserializable` class or people won't be able to load older models.
